tunnel-server/src/strings.ts

```diff
@@ -1,12 +1,13 @@
 import { createHash } from 'crypto'
 
-export const truncateWithHash = (s: string, maxLength: number, hashLength = 4, sep = '-') => {
-  if (s.length <= maxLength) {
+export const truncateWithHash = (s: string, sanitize: (s:string)=> string, maxLength: number, hashLength = 4, sep = '-') => {
+  const sanitized = sanitize(s)
+  if (s.length <= maxLength && sanitized === s) {
     return s
   }
 
   const suffix = createHash('md5').update(s).digest('base64url').replace(/[^a-zA-Z0-9]/g, '')
```

Comment:
looks like `sanitize` needs to run on the suffix as well

Reply:
Added check on the client-id sanitization + test